metadata_please/tests/pep621.py

```diff
@@ -0,0 +1,35 @@
+import tempfile
+
+import unittest
+from pathlib import Path
+
+from ..pep621 import basic_metadata_from_pep621_checkout
+from ..types import BasicMetadata
+
+
+class Pep621Test(unittest.TestCase):
+    maxDiff = None
+
+    def test_empty(self) -> None:
+        with tempfile.TemporaryDirectory() as d:
+            Path(d, "pyproject.toml").write_text("")
+            self.assertEqual(
+                BasicMetadata((), frozenset()),
+                basic_metadata_from_pep621_checkout(Path(d)),
+            )
+
+    def test_extras(self) -> None:
+        with tempfile.TemporaryDirectory() as d:
+            Path(d, "pyproject.toml").write_text(
+                """\
+[project]
+dependencies = ["x"]
+
+[project.optional-dependencies]
+dev = ["Foo <= 2"]
```

Comment:
How about adding a marker to this? 

['Foo <= 2 ;   platform_system != "Windows"']

Reply:
I added a "marker" dep as I was moving this between files, as well as adding a "y" dep with a marker re the comment above.